src/native/custom_key_op_handler.c

```diff
@@ -139,24 +157,28 @@ struct aws_jni_custom_key_op_handler *aws_custom_key_op_handler_java_new(
     java_custom_key_op_handler->key_handler.impl = (void *)java_custom_key_op_handler;
     /* Make a global reference so the Java interface is kept alive */
     java_custom_key_op_handler->jni_custom_key_op = (*env)->NewGlobalRef(env, jni_custom_key_op);
+    AWS_FATAL_ASSERT(java_custom_key_op_handler->jni_custom_key_op != NULL);
     java_custom_key_op_handler->allocator = allocator;
 
     AWS_LOGF_DEBUG(
         AWS_LS_COMMON_IO,
         "java_custom_key_op_handler=%p: Initalizing Custom Key Operations",
         (void *)java_custom_key_op_handler);
-    return java_custom_key_op_handler;
-}
 
-void aws_custom_key_op_handler_java_release(
-    struct aws_allocator *allocator,
-    struct aws_jni_custom_key_op_handler *java_custom_key_op_handler) {
+    return &java_custom_key_op_handler->key_handler;
+}
 
-    (void)allocator;
+void aws_custom_key_op_handler_java_release(struct aws_custom_key_op_handler *custom_key_op_handler) {
 
+    if (custom_key_op_handler == NULL) {
+        return;
+    }
+    struct aws_jni_custom_key_op_handler *java_custom_key_op_handler =
+        (struct aws_jni_custom_key_op_handler *)custom_key_op_handler->impl;
     if (!java_custom_key_op_handler) {
         return;
     }
```

Comment:
trivial: remove this if-check, just march on and crash

the way it is we're silently ignoring a coding error of ours